### ISSUE TITLE
git.openChange should act like a mode of editor, not a new tab

### ISSUE THREAD

**Issue by nojvek** [Microsoft/vscode#41933]
Version:          Code 1.19.2 (490ef761b76b3f3b3832eff7a588aac891e5fe80, 2018-01-10T15:49:04.682Z)
OS Version:       Darwin x64 16.6.0)

1. Currently Diff View Editor and TextEditor are opened in different tabs. I shift between two modes with 'alt+d' and 'alt+shift+d' which correspond to `git.openChange` and `git.openFile`. 
I really wish it would remain in same tab and switch in diff mode or not in diff mode. I would expect openFile and openChange to behave interchangeably.

![image](https://user-images.githubusercontent.com/1018196/35196861-b2069d62-fe8c-11e7-9e55-5074ae9bc2a0.png)

2. If you open a file, then explorer automatically scrolls to edited file. In diff mode, that functionality breaks even though it's the same file. I would expect the file to be still selected. This gives me context of where the file is relative to its parent directory and other files.
![image](https://user-images.githubusercontent.com/1018196/35196867-d2e51f90-fe8c-11e7-81b2-6c9e4decfc7e.png)

3. In file editor, I have inline change view where I can revert a specific change. I would expect this to be part of diff editor. The screen real estate of inline change editor is so small, you can barely see more than 3-4 lines unless you manually resize. Diff editor should just have an option to revert specific changes or only stage some changes. This would appear as code lens above changes

![image](https://user-images.githubusercontent.com/1018196/35196922-7b960730-fe8d-11e7-9259-15436359fbd6.png)

4. In both diff mode and inline diff mode, you can not select deleted lines. The red part of lines. It would be awesome if I can at-least select the red (deleted) portion so I can paste over green portion and make and edit. Most cases, I only want to revert a very specific portion of code and full revert is too destructive.

Basically I am proposing that the diff mode plays nicer with current editor. It should feel like you are still in the same tab, just seeing a diff perspective. Everything is selectable, and file explorer still auto-scrolls so selected file is visible.
